src/audits/performance/user-timings.js

```diff
@@ -48,36 +48,71 @@ class UserTimings extends Audit {
       });
     }
 
-    let timingsCount = 0;
+    let userTimings = [];
+    let navigationStartTime;
+    let measuresStartTimes = {};
+    let traceStart = false;
 
     // Fetch blink.user_timing events from the tracing data
     const timelineModel = new TimelineModel(artifacts.traceContents);
     const modeledTraceData = timelineModel.timelineModel();
-    const key = [...modeledTraceData.mainThreadAsyncEvents().keys()].find(
-      key => key.title === 'User Timing'
-    );
-    let userTimings = modeledTraceData.mainThreadAsyncEvents().get(key);
 
-    // Reduce events to record only useful information
-    if (typeof userTimings !== 'undefined') {
-      timingsCount = userTimings.length;
-      userTimings = userTimings.map(ut => {
-        return {
-          name: ut.name,
-          startTime: ut.startTime,
-          endTime: ut.endTime,
-          duration: ut.duration.toFixed(2) + 'ms',
-          args: ut.args
-        };
-      });
-    }
+    // Get all blink.user_timing events
+    // The event phases we are interested in are mark and instant events (R, i, I)
+    // and duration events which correspond to measures (B, b, E, e).
+    // @see https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#
+    modeledTraceData.mainThreadEvents()
+    .filter(ut => ut.hasCategory('blink.user_timing') || ut.name === 'TracingStartedInPage')
+    .forEach(ut => {
+      if (ut.phase === 'R' || ut.phase.toUpperCase() === 'I') {
+        // Mark event
+        if (ut.name === 'TracingStartedInPage' && !traceStart) {
+          traceStart = true;
+          return;
+        }
+        if (ut.name === 'navigationStart' && traceStart & !navigationStartTime) {
+          navigationStartTime = ut.startTime;
+        }
+
+        if (!ut.args.hasOwnProperty('frame') && ut.name !== 'requestStart') {
```

Comment:
yah i totally forgot why we are using frame as a heuristic for these baddies
https://cs.chromium.org/search/?q=trace_event.*?user_timing&sq=package:chromium&type=cs


lets do something dope like
```js
.filter(ut => {
  if (ut.name === 'TracingStartedInPage') || ut.name === 'navigationStart')
    return true;
  if (ut.hasCategory('blink.user_timing')){
    // reject these "userTiming" events that aren't really UserTiming, by nuking ones with frame data (or requestStart)
    // https://cs.chromium.org/search/?q=trace_event.*?user_timing&sq=package:chromium&type=cs
    return !(ut.name === 'requestStart' || ut.args.frame !== undefined);
  }
});
```